src/NuGet.Core/NuGet.Protocol/HttpSource/HttpSource.cs

```diff
@@ -435,7 +435,7 @@ public static HttpSource Create(SourceRepository source, IThrottle throttle)
                 throw new ArgumentNullException(nameof(throttle));
             }
 
-            Func<Task<HttpHandlerResource>> factory = () => source.GetResourceAsync<HttpHandlerResource>();
+            Func<Task<HttpHandlerResource>> factory = () => source.GetResourceAsync<HttpHandlerResource>(CancellationToken.None);
```

Comment:
While this is in a Client SDK package, anything that wants to make an HTTP request has to get the HttpHandlerResource, hence we can be confident that the HttpHandlerResourceProvider isn't going to be making HTTP requests (or do any other async work), so using `CancellationToken.None` here won't cause any blocking on cancellation requested.